src/blob/core.rs

```diff
@@ -316,9 +316,17 @@ impl Blob {
         self.name.id
     }
 
-    pub(crate) fn check_bloom(&self, key: &[u8]) -> Option<bool> {
-        trace!("check bloom filter");
-        self.index.check_bloom_key(key)
+    pub(crate) fn check_filters(&self, key: &[u8]) -> Option<bool> {
+        trace!("check filters (range and bloom)");
+        self.index.check_filters_key(key)
+    }
+
+    pub(crate) async fn check_filters_async<R>(&self, key: &[u8], index: R) -> Option<R> {
```

Comment:
It's a bit unintuitive to see `index: R` here... The only reason for this is `map` in `get_data_last`? Probably it's better to let it remain simple predicate and use `filter`?

Reply:
Okay, I was bitten by a crazy frog (or I don't know why I dicided to write it in that way), so now I did in a right way